src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -108,12 +109,23 @@ class MarathonScheduler @Inject() (
     val killedForFailingHealthChecks =
       status.getState == TASK_KILLED && status.hasHealthy && !status.getHealthy
 
+    lazy val maybeApp: Future[Option[AppDefinition]] = appRepo.currentVersion(appId)
+
     if (status.getState == TASK_ERROR || status.getState == TASK_FAILED || killedForFailingHealthChecks)
-      appRepo.currentVersion(appId).foreach {
+      maybeApp.foreach {
         _.foreach(taskQueue.rateLimiter.addDelay)
       }
     status.getState match {
       case TASK_ERROR | TASK_FAILED | TASK_FINISHED | TASK_KILLED | TASK_LOST =>
+        if (config.reviveOffersForNewApps()) {
```

Comment:
ScallopOption has an apply method that gets the current value. Without the `()`, it will not work.

(I do not like that API either)